lib/network/utils.ts

```diff
@@ -28,6 +29,61 @@ export function arsenalErrorKMIP(err: string | Error) {
     return _normalizeArsenalError(err, 'KMIP:');
 }
 
-export function arsenalErrorAWSKMS(err: string | Error) {
+// List of AWS KMS errors that are specific to KMS and where the error
+// can be returned to the user as is with a specific status code.
+// Avoid any error that might leak sensitive information such as hostnames
+// or ip addresses from network related errors.
+// KeyId is not considered sensitive.
+// Reference: https://docs.aws.amazon.com/kms/latest/APIReference/CommonErrors.html
+// See specific actions like CreateKey, Encrypt, Decrypt, etc. for more details.
+// Other errors not listed will return only the same error code but http status 500
+// and message will be generic and details will be in logs only.
+const allowedKmsErrors = {
```

Comment:
Shouldn't we have these errors in the https://github.com/scality/Arsenal/blob/development/8.2/lib/errors/arsenalErrors.ts

Reply:
THere is no duplication see [this](https://github.com/scality/Arsenal/blob/development/8.2/lib/errors/arsenalErrors.ts#L1076)
I think its still worth adding all errors there like the KMS one already present

Reply:
I've reworked kms errors in arsenal [97d9bfe](https://github.com/scality/Arsenal/pull/2432/commits/97d9bfefb6896ca839e88cdc53c53328a7fc0f99)